spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/ApplicationConfigurationMetadataResolverAutoConfiguration.java

```diff
@@ -80,4 +110,115 @@ public ApplicationConfigurationMetadataResolver metadataResolver(
+		registryConfigurationMap.values().stream()
+				.filter(rc -> rc.getAuthorizationType() == RegistryConfiguration.AuthorizationType.dockeroauth2)
+				.filter(rc -> !rc.getExtra().containsKey(DockerOAuth2RegistryAuthorizer.DOCKER_REGISTRY_AUTH_URI_KEY))
+				.forEach(rc -> {
+							String tokenServiceUri = secretToRegistryConfigurationConverter.getDockerTokenServiceUri(
+									rc.getRegistryHost(), rc.getUser(), rc.getSecret());
+							if (StringUtils.hasText(tokenServiceUri)) {
+								rc.getExtra().put(DockerOAuth2RegistryAuthorizer.DOCKER_REGISTRY_AUTH_URI_KEY, tokenServiceUri);
+							}
+						}
+				);
+
+		if (!StringUtils.isEmpty(dockerConfigJsonSecret)) {
+			// Retrieve registry configurations from mounted kubernetes Secret.
+			Map<String, RegistryConfiguration> secretsRegistryConfigurationMap
+					= secretToRegistryConfigurationConverter.convert(dockerConfigJsonSecret);
+
+			// Merge the Secret and the Property based registry configurations.
+			// The properties values when set has precedence over the Secret retrieved one. Later allow to override
+			// some of the Secret properties or set the disableSslVerification for secret based configs.
+			Map<String, RegistryConfiguration> mergedConfigurations = Stream.concat(
+					secretsRegistryConfigurationMap.entrySet().stream(),
+					registryConfigurationMap.entrySet().stream())
+					.collect(Collectors.toMap(
+							Map.Entry::getKey,
+							Map.Entry::getValue,
+							(secretConf, propConf) -> {
+								RegistryConfiguration rc = new RegistryConfiguration();
+								rc.setRegistryHost(secretConf.getRegistryHost());
+								rc.setUser(StringUtils.hasText(propConf.getUser()) ? propConf.getUser() : secretConf.getUser());
+								rc.setSecret(StringUtils.hasText(propConf.getSecret()) ? propConf.getSecret() : secretConf.getSecret());
+								rc.setAuthorizationType(propConf.getAuthorizationType() != null ? propConf.getAuthorizationType() : secretConf.getAuthorizationType());
+								rc.setManifestMediaType(StringUtils.hasText(propConf.getManifestMediaType()) ? propConf.getManifestMediaType() : secretConf.getManifestMediaType());
+								rc.setDisableSslVerification(propConf.isDisableSslVerification());
+								rc.getExtra().putAll(secretConf.getExtra());
+								rc.getExtra().putAll(propConf.getExtra());
+								return rc;
+							}
+					));
+			registryConfigurationMap = mergedConfigurations;
+		}
+
+		logger.info("Final Registry Configurations: " + registryConfigurationMap);
+
+		return registryConfigurationMap;
+	}
+
+	@Bean
+	public DockerConfigJsonSecretToRegistryConfigurationConverter secretToRegistryConfigurationConverter(
+			@Qualifier("noSslVerificationContainerRestTemplate") RestTemplate trustAnySslRestTemplate) {
+		return new DockerConfigJsonSecretToRegistryConfigurationConverter(trustAnySslRestTemplate);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public RestTemplate containerRestTemplate(RestTemplateBuilder builder) {
```

Comment:
The conditionalOnMissingBean works against the return type....do you want to test against the name of the bean as well?